### ISSUE TITLE
[Feature request] VSCodium support

### ISSUE THREAD

**Issue by Toub** [microsoft/vscode-pull-request-github#1309]
Installing this extension on vscodium 1.37.1.

Then, click on "Configure" icon on the new Github tab.

The following error occurs:

Notification message:

> command 'pr.configurePRViewlet' not found

Output view:

> [Info] Looking for git repository

Developer console

> [Extension Host] debugger listening on port 16659

> workbench.main.js:sourcemap:3351 Extension 'GitHub.vscode-pull-request-github cannot use PROPOSED API (must started out of dev or enabled via --enable-proposed-api)
updateEnableProposedApi @ workbench.main.js:sourcemap:3351

> Activating extension 'GitHub.vscode-pull-request-github' failed: [GitHub.vscode-pull-request-github]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.vscode-pull-request-github.
> _logMessageInConsole @ workbench.main.js:sourcemap:3348
> workbench.main.js:sourcemap:242 [Extension Host] Activating extension `GitHub.vscode-pull-request-github` failed:  [GitHub.vscode-pull-request-github]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.vscode-pull-request-github

> [Extension Host] Here is the error stack:  Error: [GitHub.vscode-pull-request-github]: 
> Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.vscode-pull-request-github
> 	at c (/usr/share/codium/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:489:844)
> 	at new t.PullRequestsTreeDataProvider (/home/toub/.vscode-oss/extensions/github.vscode-pull-request-github-0.10.0/media/extension.js:1:399642)
> 	at /home/toub/.vscode-oss/extensions/github.vscode-pull-request-github-0.10.0/media/extension.js:1:632294
> 	at Generator.next (<anonymous>)
> 	at n (/home/toub/.vscode-oss/extensions/github.vscode-pull-request-github-0.10.0/media/extension.js:1:630635)
> 	at new Promise (<anonymous>)
> 	at n (/home/toub/.vscode-oss/extensions/github.vscode-pull-request-github-0.10.0/media/extension.js:1:630412)
> 	at t.activate (/home/toub/.vscode-oss/extensions/github.vscode-pull-request-github-0.10.0/media/extension.js:1:631897)
> 	at Function._callActivateOptional (/usr/share/codium/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:764:518)
> 	at Function._callActivate (/usr/share/codium/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:764:207)
> 	at define._doActivateExtension.Promise.all.then.e (/usr/share/codium/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:763:245)

> notificationsAlerts.ts:38 Error: command 'pr.configurePRViewlet' not found

Is this extension supposed to work on vscodium, and is there a chance it is supported in the future?

**Comment by RMacfarlane** [comment]
Hey @Toub, yes, the extension should work on vscodium, but you do have to provide an additional flag when launching. Using `code --enable-proposed-api GitHub.vscode-pull-request-github` from the command line should work.

The reason for this is that in the product build of VSCode, the product.json gets updated to whitelist some extensions as being allowed to use the proposed API, and this isn't in the build of VSCodium. Once all parts of the API we're using are no longer proposed, it will work without the flag.

**Comment by RMacfarlane** [comment]
I created an issue to track removing the command line switch: https://github.com/microsoft/vscode-pull-request-github/issues/1310

**Comment by Toub** [comment]
Ok great. Thanks for the clarification!

**Comment by RMacfarlane** [comment]
Closing this since you're unblocked and there's another issue to track the removal of the command line switch. Thanks!

**Comment by 7automaton** [comment]
Hello,
I use Codium. The authentication API doesn't like the `vscodium://` URI, contrary to the `vscode://` URI, as shown in this GIF:

![vysCCNF8Cp](https://user-images.githubusercontent.com/56928485/81430002-50e11580-9124-11ea-893f-272079a63537.gif)

The API will let me through with the `vscode://` API with an `801` code (I don't have VS Code installed), and it forbids me if I use the `vscodium://` URI with `Forbidden`.

This is the link to authenticate, with state and windowId redacted:

`https://vscode-auth.github.com/authorize/?callbackUri=vscodium://github.vscode-pull-request-github/did-authenticate%253F...&scope=read:user%20user:email%20repo%20write:discussion&responseType=code&authServer=https://github.com/`

The callback URI differed:
```
✔️ (801)       ?callbackUri=vscode://github.vscode-pull-request-github/did-authenticate
❌ (Forbidden) ?callbackUri=vscodium://github.vscode-pull-request-github/did-authenticate
```

If it helps, I'm using Codium as a portable binary, but I opened the link with it running in the background.

**Comment by 7automaton** [comment]
Actually, please don't respond to this yet. I'm going to move this to a new issue.

**Comment by scottwallacesh** [comment]
> Hey @Toub, yes, the extension should work on vscodium, but you do have to provide an additional flag when launching. Using `code --enable-proposed-api GitHub.vscode-pull-request-github` from the command line should work.

To prevent you running from the command line each time, you can add an entry to `~/.vscode-oss/argv.json`:
```javascript
{
    // Allow Github Pull Request to work
    "enable-proposed-api": [
        "GitHub.vscode-pull-request-github"
    ]
}
```

**Comment by JamesS-M** [comment]
> > Hey @Toub, yes, the extension should work on vscodium, but you do have to provide an additional flag when launching. Using `code --enable-proposed-api GitHub.vscode-pull-request-github` from the command line should work.
> 
> To prevent you running from the command line each time, you can add an entry to `~/.vscode-oss/argv.json`:
> 
> ```js
> {
>     // Allow Github Pull Request to work
>     "enable-proposed-api": [
>         "GitHub.vscode-pull-request-github"
>     ]
> }
> ```

@scottwallacesh Thank you so much for this!
